src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -35,7 +36,27 @@ public class RestoreCommand
 
         public Guid ParentId { get; }
 
-        public const string ProjectRestoreInformation = "ProjectRestoreInformation";
+        // names for ProjectRestoreInformation and intervals
+        private static readonly string ProjectRestoreInformation = nameof(ProjectRestoreInformation);
+        private static readonly string Errors = nameof(Errors);
+        private static readonly string ErrorMessages = nameof(ErrorMessages);
+        private static readonly string Warnings = nameof(Warnings);
+        private static readonly string WarningMessages = nameof(WarningMessages);
+        private static readonly string Success = nameof(Success);
+
+        // names for child events for ProjectRestoreInformation
+        private static readonly string ExecuteRestoreAsyncInformation = nameof(ExecuteRestoreAsyncInformation);
+        private static readonly string BuildAssetsFileInformation = nameof(BuildAssetsFileInformation);
+        private static readonly string ValidateRestoreGraphsAsyncInformation = nameof(ValidateRestoreGraphsAsyncInformation);
+        private static readonly string VerifyCompatibilityAsyncInformation = nameof(VerifyCompatibilityAsyncInformation);
+        private static readonly string CreateRestoreResultInformation = nameof(CreateRestoreResultInformation);
+        private static readonly string NoOpInformation = nameof(NoOpInformation);
+
+        // names for intervals in NoOpInformation
+        private static readonly string NoOpEvaluateCacheFileDuration = nameof(NoOpEvaluateCacheFileDuration);
+        private static readonly string NoOpSuccessEvaluationDuration = nameof(NoOpSuccessEvaluationDuration);
+        private static readonly string NoOpSuccessEvaluation = nameof(NoOpSuccessEvaluation);
+        private static readonly string NoOpReplayLogsDuration = nameof(NoOpReplayLogsDuration);
```

Comment:
what is the purpose of this field? again is this significant enough to be logged as separate field? or do we just want to log duration for each line of code?

Reply:
Logging can be an expensive operation and it helps to know how much time is being lost because of logging.